src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -349,15 +361,40 @@ public void callAllGetOffset() {
 
   @Override
   public void closeAll() {
+    if (closeChannelsInParallel) {
+      closeAllInParallel();
+    } else {
+      closeAllSequentially();
+    }
+
+    partitionsToChannel.clear();
+
+    StreamingClientProvider.getStreamingClientProviderInstance()
+        .closeClient(this.connectorConfig, this.streamingIngestClient);
+  }
+
+  private void closeAllSequentially() {
     partitionsToChannel.forEach(
         (partitionChannelKey, topicPartitionChannel) -> {
           LOGGER.info("Closing partition channel:{}", partitionChannelKey);
           topicPartitionChannel.closeChannel();
         });
-    partitionsToChannel.clear();
+  }
 
-    StreamingClientProvider.getStreamingClientProviderInstance()
-        .closeClient(this.connectorConfig, this.streamingIngestClient);
+  private void closeAllInParallel() {
+    CompletableFuture<?>[] futures =
+        partitionsToChannel.entrySet().stream()
+            .map(
+                entry -> {
+                  String channelKey = entry.getKey();
+                  TopicPartitionChannel topicPartitionChannel = entry.getValue();
+
+                  LOGGER.info("Closing partition channel:{}", channelKey);
+                  return topicPartitionChannel.closeChannelAsync();
+                })
+            .toArray(CompletableFuture[]::new);
+
+    CompletableFuture.allOf(futures).join();
   }
```

Comment:
In my small comprehension,  `CompletableFuture` uses the `ForkJoinPool.commonPool()` as its default thread pool with `(Runtime.getRuntime().availableProcessors() - 1)` threads. We can cheat by setting JVM cpu count but I rather suggest to create a thread pools like this:

```
import java.util.concurrent.*;

private void closeAllInParallel() {
    ExecutorService executor = Executors.newFixedThreadPool(CONFIG_THREAD_POOL_SIZE);
    CompletableFuture<?>[] futures =
        partitionsToChannel.entrySet().stream()
            .map(
                entry -> {
                  String channelKey = entry.getKey();
                  TopicPartitionChannel topicPartitionChannel = entry.getValue();

                  LOGGER.info("Closing partition channel:{}", channelKey);
                  return CompletableFuture.runAsync(topicPartitionChannel::closeChannelAsync, executor);
                })
            .toArray(CompletableFuture[]::new);

    CompletableFuture.allOf(futures).join();
    executor.shutdown(); // Don't forget to shutdown the executor
}
```

Reply:
Thanks, @ebuildy, for providing an example. Let me merge the PR in its current form, I'll check with the SDK team on a possible built-in solution. If the timeline is not optimistic, I'll propose this change to the rest of the team.

Reply:
fantastic! Any plan to do the same for open channel operations?